Intersections_2/include/CGAL/Intersections_2/Segment_2_Segment_2.h

```diff
@@ -274,10 +274,10 @@ do_intersect(const typename K::Segment_2 &seg1,
 template <class K>
 class Segment_2_Segment_2_pair {
 public:
-    enum Intersection_results {NO_INTERSECTION, POINT, SEGMENT};
+    enum Intersection_results {NO_INTERSECTION, POINT, SEGMENT, UNKNOWN};
     Segment_2_Segment_2_pair(typename K::Segment_2 const *seg1,
                             typename K::Segment_2 const *seg2)
-            : _seg1(seg1), _seg2(seg2), _known(false) {}
+            : _seg1(seg1), _seg2(seg2), _known(false), _result(UNKNOWN) {}
```

Comment:
It seems the code assumes:
```c++
assert(_known == (_result != UNKNOWN));
```
as an invariant. According to the [Don't repeat yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle, the Boolean variable `_known` should be removed.

Reply:
@lrineau I agree. The pattern of using `_known` seems to be present in several other places and is always initialized, whereas `_result` isn't. I think the pattern of using an 'Unkown' enum value is simpler and more clearly understood. But it would be nice as part of this PR to make it consistent across all. I can do so if wanted.

Reply:
I'll take care of it.